src/java.base/share/classes/java/lang/foreign/ValueLayout.java

```diff
@@ -104,9 +104,10 @@ public ValueLayout withOrder(ByteOrder order) {
 
     @Override
     public String toString() {
-        return decorateLayoutString(String.format("%s%d",
+        return decorateLayoutString(String.format("%s%d(%s)",
                 order == ByteOrder.BIG_ENDIAN ? "B" : "b",
-                bitSize()));
+                bitSize(),
+                carrier == MemoryAddress.class ? "MA" : carrier.descriptorString()));
```

Comment:
I have thought separately about that. I think I'd like to change this so that we drop the `b` and we replace it with a one letter descriptor (as per descriptor string), and use `a` for `MemoryAddress. Then, as for `b` we use upper case for bigg-endian, and lower-case for little endian. That way the toString representation is compact in all cases (while with this patch it blows up on MemoryAddress).

Reply:
Good suggestion. I went with this scheme since it seemed more extensible if we wanted to add more carriers in the future. But, it's not like the toString format is specified, so I guess we can change it up later if needed.